src/replication.c

```diff
@@ -2225,6 +2266,12 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 		if (rc) {
 			if (rc == 1) {
 				rc = cmd->data_len = rcomm_cmd->data_len;
+			} else if (rcomm_cmd->opcode == ZVOL_OPCODE_READ &&
+			    rcomm_cmd->copies_sent == 1) {
+				rcomm_cmd->copies_sent = 0;
+				memset(rcomm_cmd->resp_list, 0, sizeof (rcomm_cmd->resp_list));
+				MTX_LOCK(&spec->rq_mtx);
```

Comment:
should we reset cmd_sent local variable also?

Reply:
IO may be going on with few other replicas which are slow, and, may overwrite rcomm_cmd->resp_list with the index given in its rcmd_t (replica_cmd). This can lead to issues.

Reply:
num_read_ios local variable also need to reset